chat-plugins/info.js

```diff
@@ -140,37 +140,24 @@ exports.commands = {
 		}
 
 		if (user.can('alts', targetUser) || (room.isPrivate !== true && user.can('mute', targetUser, room) && targetUser.userid in room.users)) {
-			let roomPunishments = ``;
-			for (let i = 0; i < Rooms.global.chatRooms.length; i++) {
-				const curRoom = Rooms.global.chatRooms[i];
-				if (!curRoom || curRoom.isPrivate === true) continue;
-				let punishment = Punishments.roomIps.nestedGet(curRoom.id, targetUser.latestIp);
-				if (!punishment) {
-					punishment = Punishments.roomUserids.nestedGet(curRoom.id, targetUser.userid);
-				}
-				let punishDesc = ``;
-				if (punishment) {
-					const [punishType, userid, expireTime, reason] = punishment;
-					punishDesc = Punishments.roomPunishmentTypes.get(punishType);
-					if (!punishDesc) punishDesc = `punished`;
-					if (userid !== targetUser.userid) punishDesc += ` as ${userid}`;
+			let punishments = Punishments.getRoomPunishments(user);
+
+			if (punishments.length) {
```

Comment:
``punishments`` is USUALLY undefined here, so for a user whose not punished, this will crash.